packages/flutter_tools/test/general.shard/flutter_manifest_test.dart

```diff
@@ -648,82 +810,117 @@ flutter:
+      Uri.parse('lib/gallery/aaa%20bbb'),
+    ]);
+  });
 
-    testUsingContext('Returns proper error when flutter is a list instead of a map', () async {
-      const String manifest = '''
+  testWithoutContext('FlutterManifest returns proper error when flutter is a '
+    'list instead of a map', () {
+    const String manifest = '''
 name: test
 dependencies:
   flutter:
     sdk: flutter
 flutter:
   - uses-material-design: true
 ''';
-      final FlutterManifest flutterManifest = FlutterManifest.createFromString(manifest);
+    final BufferLogger logger = BufferLogger.test();
+    final FlutterManifest flutterManifest = FlutterManifest.createFromString(
+      manifest,
+      logger: logger,
+    );
 
-      expect(flutterManifest, null);
-      expect(testLogger.errorText, contains('Expected "flutter" section to be an object or null, but got [{uses-material-design: true}].'));
-    });
+    expect(flutterManifest, null);
+    expect(logger.errorText,
+      contains(
+        'Expected "flutter" section to be an object or null, but got '
+        '[{uses-material-design: true}].',
+      ),
+    );
   });
 
-  group('FlutterManifest with MemoryFileSystem', () {
-    Future<void> assertSchemaIsReadable() async {
-      const String manifest = '''
+  testWithoutContext('FlutterManifest with MemoryFileSystem', () {
+    const String manifest = '''
 name: test
 dependencies:
   flutter:
     sdk: flutter
 flutter:
 ''';
-
-      final FlutterManifest flutterManifest = FlutterManifest.createFromString(manifest);
-      expect(flutterManifest.isEmpty, false);
-    }
-
-    void testUsingContextAndFs(
-      String description,
-      FileSystem filesystem,
-      dynamic testMethod(),
-    ) {
-      testUsingContext(
-        description,
-        () async {
-          writeEmptySchemaFile(filesystem);
-          testMethod();
-        },
-        overrides: <Type, Generator>{
-          FileSystem: () => filesystem,
-          ProcessManager: () => FakeProcessManager.any(),
-        },
-      );
-    }
-
-    testUsingContext('Validate manifest on original fs', () {
-      assertSchemaIsReadable();
-    });
-
-    testUsingContextAndFs(
-      'Validate manifest on Posix FS',
-      MemoryFileSystem(style: FileSystemStyle.posix),
-      () {
-        assertSchemaIsReadable();
-      },
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    fileSystem.file('pubspec.yaml').writeAsStringSync(manifest);
+    final BufferLogger logger = BufferLogger.test();
+    final FlutterManifest flutterManifest = FlutterManifest.createFromPath(
+       'pubspec.yaml',
+      fileSystem: fileSystem,
+      logger: logger,
     );
 
-    testUsingContextAndFs(
-      'Validate manifest on Windows FS',
-      MemoryFileSystem(style: FileSystemStyle.windows),
-      () {
-        assertSchemaIsReadable();
-      },
+    expect(flutterManifest.isEmpty, false);
+  });
+
+  testWithoutContext('FlutterManifest can parse manifest on posix filesystem', () {
```

Comment:
you mean windows?

Reply:
Fixed, good catch

Reply:
Hopefully that makes up for me taking so long to review it :P